packages/engine-server/src/workspace.ts

```diff
@@ -199,15 +199,38 @@ export class WorkspaceService {
     return;
   }
 
+  shouldWorkspaceVaultSync(
+    command: "commit" | "push" | "pull",
+    vaultRoot: string
+  ): boolean {
+    if (!WorkspaceService.isWorkspaceVault(vaultRoot)) return true;
+    let config = this.config.workspaceVaultSync;
+    if (_.isUndefined(config)) config = "noPush"; // default
```

Comment:
I thought default should be noCommit? According to https://github.com/dendronhq/dendron/pull/789/files#diff-c34b22187b5e218639151cc49620e31d30fea1a5ababad8d17b8f9bf7efc3249R169

Reply:
Oh right, thanks for the reminder. I was actually going to ask what the default should be. I really like `noCommit`, but `noPush` preserves the current behavior. Any thoughts?

Reply:
I thought current default is "sync". You have to work to add things to private and not have it sync by default. I *think* the simplest would actually be to default to sync and have the other options be available. Especially since it's called the workspace sync command and that's the behavior we'd like to encourage.

Reply:
default should be commit and push unless otherwise specified by config

Reply:
This configuration is only for [workspace vaults](https://wiki.dendron.so/notes/c6fd6bc4-7f75-4cbb-8f34-f7b99bfe2d50.html#workspace-vault). The previous behavior for workspace vaults was the same as `noPush`, this pull request makes that configurable. Regular vaults will continue to get the `sync` behavior and are not configurable with this pull request.

Reply:
I'll add a per-vault configuration as @kevinslin suggested, which will apply to all vaults. This config option will continue to only affect workspace vaults, unless someone suggests otherwise.

Reply:
@kevinslin Just to avoid any miscommunication: previous behavior for workspace vaults was to commit and pull but [not push](https://github.com/dendronhq/dendron/blob/6f401d75f21122c84efd03f4307531fde719e37d/packages/engine-server/src/workspace.ts#L446). Should I be changing that behavior? To which option?